homeassistant/components/sonos/media_browser.py

```diff
@@ -46,6 +46,32 @@
 type GetBrowseImageUrlType = Callable[[str, str, str | None], str]
 
 
+def fix_image_url(url: str) -> str:
+    """Update the image url to fully encode characters to allow image display in media_browser UI.
+
+    Images whose file path contains characters such as ',()+ are not loaded without escaping them.
+    """
+
+    # Before parsing encode the plus sign; otherwise it'll be interpreted as a space.
+    original_uri: str = urllib.parse.unquote(url).replace("+", "%2B")
+    parsed_url = urllib.parse.urlparse(original_uri)
+    query_params = urllib.parse.parse_qsl(parsed_url.query)
+    new_url = urllib.parse.urlunsplit(
+        (
+            parsed_url.scheme,
+            parsed_url.netloc,
+            parsed_url.path,
+            urllib.parse.urlencode(
+                query_params, quote_via=urllib.parse.quote, safe="/:"
+            ),
+            "",
+        )
+    )
+    if original_uri != new_url:
```

Comment:
```suggestion
    # Before parsing encode the plus sign; otherwise it'll be interpreted as a space.
    original_url: str = urllib.parse.unquote(url).replace("+", "%2B")
    parsed_url = urllib.parse.urlparse(original_url)
    query_params = urllib.parse.parse_qsl(parsed_url.query)
    new_url = urllib.parse.urlunsplit(
        (
            parsed_url.scheme,
            parsed_url.netloc,
            parsed_url.path,
            urllib.parse.urlencode(
                query_params, quote_via=urllib.parse.quote, safe="/:"
            ),
            "",
        )
    )
    if original_url != new_url:
```